packages/devtools/lib/src/settings/settings_screen.dart

```diff
@@ -39,43 +44,141 @@ class SettingsScreen extends Screen {
+
+  Spinner _loadingVersionSpinner;
+
   SettingsController _controller;
 
   @override
   CoreElement createContent(Framework framework) {
     ga_platform.setupDimensions();
     this.framework = framework;
-    _screenDiv = div(c: 'custom-scrollbar')..layoutVertical();
-    _container = div(c: 'settings-container');
-    _screenDiv
-      ..clear()
-      ..add(_container..clear());
-    _container.add(_sdkVersion);
-    _container.add(_flagList);
+
+    _initContent();
+    _checkForFlutterVersionService();
+
+    final CoreElement screenDiv = div(c: 'custom-scrollbar')..layoutVertical();
+
+    screenDiv
+      ..add([
+        div(c: 'section')
+          ..add([
+            h2(text: 'Version Information'),
+            _versionContainer,
+          ]),
+        div(c: 'section')
+          ..flex()
+          ..add([
+            h2(text: 'Dart VM Flag List'),
+            div(c: 'flag-list-container')
+              ..flex()
+              ..add(_flagList),
+          ])
+      ]);
 
     _controller.entering();
-    return _screenDiv;
+    return screenDiv;
+  }
+
+  void _initContent() {
+    _flagList = div(c: 'flag-list')..layoutVertical();
+
+    _versionContainer = div(c: 'version-container')
+      ..layoutVertical()
+      ..flex()
+      ..add([
+        _loadingVersionSpinner = Spinner.centered(),
+        _sdkVersionContainer = div()
+          ..layoutHorizontal()
+          ..add([
+            h5(text: 'Dart SDK:'),
+            _sdkVersion = h5(c: 'version-value'),
+          ])
+          ..hidden(true),
+        _flutterVersionContainer = div()
+          ..layoutVertical()
+          ..hidden(true)
+      ]);
+  }
+
+  void _checkForFlutterVersionService() async {
+    // Wait for a small delay to give DevTools a chance to pick up the
+    // registered service. This prevents the version UI from flickering.
+    await Future.delayed(const Duration(seconds: 1));
+
+    if (await serviceManager.connectedApp.isAnyFlutterApp) {
+      serviceManager.hasRegisteredService(
+        registrations.flutterVersion.service,
+        (bool serviceAvailable) async {
+          if (serviceAvailable) {
+            final FlutterVersion version =
+                await _controller.getFlutterVersion();
+            _updateVersionUI(useFlutterVersionData: true, version: version);
+          } else {
+            _updateVersionUI(useFlutterVersionData: false);
+          }
+        },
+      );
+    } else {
+      _updateVersionUI(useFlutterVersionData: false);
+    }
+    _loadingVersionSpinner.remove();
+  }
+
+  void _updateVersionUI({
+    @required bool useFlutterVersionData,
+    FlutterVersion version,
+  }) {
+    if (useFlutterVersionData && version != null) {
+      _flutterVersionContainer
+        ..clear()
+        ..add([
+          div()
+            ..layoutHorizontal()
+            ..add([
+              h5(text: 'Flutter:'),
+              h5(c: 'version-value', text: version.flutterDisplay),
```

Comment:
nit: this aren't really headings so instead use a css class with the desired style for these names and values.